pkg/controller/cincinnati/cincinnati_controller.go

```diff
@@ -214,39 +234,117 @@ func handleCACertStatus(reqLogger logr.Logger, status *cv1beta1.CincinnatiStatus
+		handleErr(reqLogger, &instance.Status, "PullSecretNotFound", err)
+		return nil, err
+	} else if err != nil {
+		return nil, err
+	}
+
+	return pullSecret, nil
+}
+
+// findTrustedCAConfig - Locate the ConfigMap referenced by the ImageConfig resource in openshift-config and return it
+func (r *ReconcileCincinnati) findTrustedCAConfig(ctx context.Context, reqLogger logr.Logger, instance *cv1beta1.Cincinnati, resources *kubeResources) (*corev1.ConfigMap, error) {
+
 	// Check if the Cluster is aware of a registry requiring an
 	// AdditionalTrustedCA
 	image := &apicfgv1.Image{}
 	err := r.client.Get(ctx, types.NamespacedName{Name: defaults.ImageConfigName, Namespace: ""}, image)
 	if err != nil && errors.IsNotFound(err) {
 		m := fmt.Sprintf("image.config.openshift.io not found for (Name: %v, Namespace: %v)", defaults.ImageConfigName, "")
-		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
-		return nil
+		handleCACertStatus(reqLogger, &instance.Status, "FindAdditionalTrustedCAFailed", m)
+		return nil, nil
 	} else if err != nil {
-		return err
+		return nil, err
 	}
 
 	if image.Spec.AdditionalTrustedCA.Name == "" {
 		m := fmt.Sprintf("image.config.openshift.io.Spec.AdditionalTrustedCA.Name not found for image (Name: %v, Namespace: %v)", defaults.ImageConfigName, "")
-		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
-		return nil
+		handleCACertStatus(reqLogger, &instance.Status, "FindAdditionalTrustedCAFailed", m)
+		return nil, nil
 	}
 
 	// Search for the ConfigMap in openshift-config
 	sourceCM := &corev1.ConfigMap{}
 	err = r.client.Get(ctx, types.NamespacedName{Name: image.Spec.AdditionalTrustedCA.Name, Namespace: openshiftConfigNamespace}, sourceCM)
 	if err != nil && errors.IsNotFound(err) {
 		m := fmt.Sprintf("Found image.config.openshift.io.Spec.AdditionalTrustedCA.Name but did not find expected ConfigMap (Name: %v, Namespace: %v)", image.Spec.AdditionalTrustedCA.Name, openshiftConfigNamespace)
-		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
-		return err
+		handleCACertStatus(reqLogger, &instance.Status, "FindAdditionalTrustedCAFailed", m)
+		return nil, err
 	} else if err != nil {
-		return err
+		return nil, err
 	}
 
 	if _, ok := sourceCM.Data[NameCertConfigMapKey]; !ok {
 		m := fmt.Sprintf("Found ConfigMap referenced by ImageConfig.Spec.AdditionalTrustedCA.Name but did not find key 'cincinnati-registry' for registry CA cert in ConfigMap (Name: %v, Namespace: %v)", image.Spec.AdditionalTrustedCA.Name, openshiftConfigNamespace)
 		handleCACertStatus(reqLogger, &instance.Status, "EnsureAdditionalTrustedCAFailed", m)
+		return nil, nil
+	}
+
+	return sourceCM, nil
+}
+
+func (r *ReconcileCincinnati) ensurePullSecret(ctx context.Context, reqLogger logr.Logger, instance *cv1beta1.Cincinnati, resources *kubeResources) error {
+	sourcePS, err := r.findPullSecret(ctx, reqLogger, instance, resources)
```

Comment:
Isn't this going to find the one in openshift's config namespace? I also see that it's not using this value later. Do we need to do this at all here?

Reply:
Yes, this is all for error checking.  We need to answer the question:
    _Is there a non empty pull-secret in openshift-config?_
   **No** -> error.  There should always be a secret.
   **Yes** and **secret == nil** -> there's nothing to copy (user doesn't have registry auth).  I can see a case for this being an error.  However, this is a weird state that I don't know how the cluster is functional at all, so I'm inclined to leave it.
   **Yes** and **secret != nil** -> continue.  Ensure the pull secret is in the local ns

Reply:
I'm not sure what you mean here by `secret == nil`. What value are you referring to exactly?

It seems like if reconciliation got this far, then it did find a pull secret in openshift-config, and it already used that to generate the desired state for a local copy of that secret. Once we get here, why do we need to re-check for the original? Why not just proceed with ensuring the desired state that was previously decided upon?

Reply:
I'm assuming the GET can return an empty secret where as `secret == nil`.  

Right, we answered the question: _Is there a non empty pull-secret in openshift-config?_ already in `postAddPullSecret`. So the question is whether there is a point to checking it a second time.  The only value I see is if a user modified the pull-secret in the time between function calls.   I think that's pretty unlikely and we would _eventually_ catch it.